Formula/go.rb

```diff
@@ -8,6 +8,17 @@ class Go < Formula
     version "1.6.3"
     sha256 "6326aeed5f86cf18f16d6dc831405614f855e2d416a91fd3fdc334f772345b00"
 
+    # 1.6.3 does not build on macOS Sierra. Users should use devel instead
+    # until 1.7 is stable (due soon).
+    depends_on MaximumMacOSRequirement => :el_capitan
+
+    # Should use the last stable binary release to bootstrap.
+    resource "gobootstrap" do
+      url "https://storage.googleapis.com/golang/go1.6.2.darwin-amd64.tar.gz"
+      version "1.6.2"
```

Comment:
Technically speaking, I would presume there's no reason why you couldn't bootstrap with the current release, but I haven't tested this, and it's useful to bootstrap with a version we _know_ works as expected and didn't cause any compile bugs. That way we should always know any compile bugs were introduced with the latest release.

Another minor consideration is political, to an extent. I wanted to avoid the _"If you can use the binary for the current release to bootstrap, why can't you just use the binary completely rather than wasting my life building from source"_ kind of discussion that pops up every now & again.